### PR TITLE
Feature/atlantis service

### PR DESCRIPTION
**Pull request by majormoses**
#### General

~- [ ] Update Changelog following the conventions laid out [here](https://github.com/sensu-plugins/community/blob/master/HOW_WE_CHANGELOG.md)~

- [x] Update README with any necessary configuration snippets

- [x] Existing tests pass

#### New Features

- [x] Tests

~- [ ] Added to the README~

#### Purpose
Have a functional service (currently only supporting `systemd` but plans to support others are a must fo rmy org. This includes setting up users, groups, config, services, etc.

#### Known Compatibility Issues
none

### CODE REVIEW

**Comment by huynt1979** [comment]
@majormoses I'll review it later on today.

File: resources/atlantis_service_systemd.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      },
+      'Service' => {
+        'Type' => 'simple',
+        'ExecStart' => "#{new_resource.atlantis_bin_location} server --config #{new_resource.atlantis_home}/#{new_resource.atlantis_config_name}",
+        'ExecStop' => '/bin/kill $MAINPID',
+        'TimeoutStopSec' => new_resource.timeout_stop_sec.to_i,
+        'SuccessExitStatus' => new_resource.atlantis_success_exit_status.to_i, # support strings
+        'User' => new_resource.atlantis_user,
+        'Group' => new_resource.atlantis_group,
+        'Environment' => 'PATH=$PATH:/usr/local/bin/terraform',
```
Comment (huynt1979) [inline resources/atlantis_service_systemd.rb]: I think terraform path should be a resource property

File: resources/atlantis_installer.rb [hunk 2]
```diff
@@ -20,13 +19,18 @@
   # install atlantis
   ark 'atlantis' do
```
Comment (huynt1979) [inline resources/atlantis_installer.rb]: I'm thinking putting atlantis and terraform in /opt as follows: `ark 'atlantis' do
  url <<url>>
  version <<version>>
  checksum <<checksum>>
  prefix_root '/opt'
  prefix_home '/opt'
  has_binaries ['atlantis']
  strip_components 0
  owner <<user>>
  group <<group>>
end`


This way we can retain multiple versions if we want to. Also, with has_binaries, ark will automatically symlink to /usr/local/bin/

File: resources/atlantis_config.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+resource_name :atlantis_config
```
Comment (tas50) [inline resources/atlantis_config.rb]: just name the file config and avoid the need to set the name

File: resources/atlantis_service_systemd.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+property :atlantis_bin_location, String, default: '/usr/local/bin/atlantis'
+property :atlantis_config_name, String, default: 'atlantis.yaml'
+property :atlantis_group, String, default: 'atlantis'
+property :atlantis_home, String, default: '/opt/atlantis'
+property :timeout_stop_sec, [Integer, String ], default: 5
+# The exit status code 143 = 128 + 15 = default terminate by system when the application doesn't have one
+property :atlantis_success_exit_status, [Integer, String], default: 143
+property :atlantis_user, String, default: 'atlantis'
+
+default_action :setup
```
Comment (tas50) [inline resources/atlantis_service_systemd.rb]: no need to define this. The first one is the default in custom resources

**Pull request merged**